std/root.ln

```diff
@@ -2583,26 +2583,83 @@ export fn prune(n: Node<any>): Tree<any> {
   return n.tree;
 }
 
+// Takes the given node and copies that node and all of its children into a new tree with the
+// current node as the root. Unfortunately a sequential algorithm right now.
+export fn toSubtree(n: Node<any>): Tree<any> {
+  const val = n.tree.vals[n.id].getR();
+  let outTree = newTree(val);
+  let children = n.tree.children[n.id].getOr(new Array<int64> []);
+  if children.length().gt(0) {
+    let parentIds = [0].repeat(children.length());
+    seqdo(newseq(pow(2, 64).getOr(0)), fn (): bool {
+      const childId = children.pop();
+      const parentId = parentIds.pop();
+      const childVal = n.tree.vals[childId].getR();
+      const grandchildIds = n.tree.children[childId].getOr(new Array<int64> []);
+      const newParentId = outTree.vals.length();
+      outTree.vals.push(childVal);
+      outTree.parents.push(parentId);
+      outTree.children.push(new Array<int64> []);
+      children = grandchildIds.concat(children);
+      parentIds = [newParentId].repeat(grandchildIds.length()).concat(parentIds);
+      return parentIds.length().gt(0);
+    });
+  }
+  return outTree;
+}
+
 export fn getChildren(t: Tree<any>): Array<Node<any>> = t.getRootNode().getChildren();
 
-// returns the new child node added
+// Attaches the value as a child of the specified node. Returns the new child node added for easier
+// method chaining
 export fn addChild(n: Node<any>, val: any): Node<any> {
-  const childId = length(n.tree.vals);
-  push(n.tree.vals, val);
-  push(n.tree.parents, n.id);
-  push(n.tree.children, new Array<int64> [ ]);
-  push(getOr(n.tree.children[n.id], new Array<int64> []), childId);
+  const childId = n.tree.vals.length();
+  n.tree.vals.push(val);
+  n.tree.parents.push(n.id);
+  n.tree.children.push(new Array<int64> []);
+  n.tree.children[n.id].getOr(new Array<int64> []).push(childId);
   return new Node<any> {
     id: childId,
     tree: n.tree,
   };
 }
 
+// Attaches the specified Tree to the specified Node. Can be done with simple maps and concats. This
+// returns the new child node corresponding to the root node of the tree.
+export fn addChild(n: Node<any>, val: Tree<any>): Node<any> {
```

Comment:
what happens if `val` and `n` are already in the same tree?

Reply:
Since we map the nodes of the tree into new arrays before concatenating them with the original arrays and replacing them, it *should* work, actually.

Reply:
Do I need to write a test for that? :/

Reply:
> Since we map the nodes of the tree into new arrays before concatenating them with the original arrays and replacing them, it _should_ work, actually.

Wait, what I meant is that if they are in the same tree already and we create another relationship between them we would create a cycle in the tree

Reply:
Oh, no, when a tree is attached to another tree, we're basically copying everything over from the one tree to the other and restitching the parent/child relationships, so there's no cycle here.

We effectively take the first tree, sorta clone it, then attach that clone as a child of whatever node we attached it to. Eg:

```
    A          A                                                                                       
   / \   =>   / \                                                                                      
  B   C      B   C                                                                                     
                 |                                                                                     
                 A'                                                                                    
                / \                                                                                    
               B'  C'
```

Reply:
> Oh, no, when a tree is attached to another tree, we're basically copying everything over from the one tree to the other and restitching the parent/child relationships, so there's no cycle here.
> 
> We effectively take the first tree, sorta clone it, then attach that clone as a child of whatever node we attached it to. Eg:
> 
> ```
>     A          A                                                                                       
>    / \   =>   / \                                                                                      
>   B   C      B   C                                                                                     
>                  |                                                                                     
>                  A'                                                                                    
>                 / \                                                                                    
>                B'  C'
> ```

ohh got it, can we include your ASCII Art prowess in the method description comment? :)

Reply:
You want this diagram added to the comments?

Reply:
Yes!